daemon/src/wire.rs

```diff
@@ -50,8 +50,32 @@ pub type Write<D, E> = SplitSink<Framed<TcpStream, EncryptedJsonCodec<D, E>>, E>
 pub struct Version(semver::Version);
 
 impl Version {
+    const CURRENT: semver::Version = semver::Version::new(2, 0, 0);
+
     pub fn current() -> Self {
-        Self(semver::Version::new(2, 0, 0))
+        Self(Self::CURRENT)
+    }
+
+    pub fn is_compatible(taker: &Self, maker: &Self) -> bool {
+        let taker = &taker.0;
+        let maker = &maker.0;
+
+        if taker.major != maker.major {
+            // major versions don't match
+            return false;
+        }
+
+        if taker.minor > maker.minor {
+            // taker's minor version is ahead of taker, that is not allowed
+            return false;
+        }
+
+        if taker.patch > maker.patch {
```

Comment:
I would consider relaxing the patch requirement - the patch versions ought to be compatible if we're daring to use SemVer here

Reply:
I agree that we should relax it for patch. I added a fixup that removes the check for patch.

We gave `semver`s `VersionReq` (as in the requirement for comparison) a try as well, but I feel it just made the code more complex. I feel the current logic is quite easy to understand and we can quite easily change the conditions.